lib/grpc_reflection/service/builder.ex

```diff
@@ -198,4 +279,102 @@ defmodule GrpcReflection.Service.Builder do
   end
 
   defp upcase_first(<<first::utf8, rest::binary>>), do: String.upcase(<<first::utf8>>) <> rest
+
+  # generate a field descriptor from a field props struct
+  defp convert_to_field_descriptor(
+         extendee,
+         %Protobuf.Extension.Props.Extension{field_props: field_props}
+       ) do
+    {type, type_name} = type_from_field_props(field_props)
+
+    %Google.Protobuf.FieldDescriptorProto{
+      name: field_props.name,
+      number: field_props.fnum,
+      label: label_from_field_props(field_props),
+      type: type,
+      type_name: type_name,
+      extendee: extendee
+    }
+  end
+
+  # Google.Protobuf.FieldDescriptorProto.Label
+  defp label_from_field_props(%Protobuf.FieldProps{optional?: true}), do: 1
+  defp label_from_field_props(%Protobuf.FieldProps{repeated?: true}), do: 3
+  defp label_from_field_props(%Protobuf.FieldProps{required?: true}), do: 2
+
+  # Google.Protobuf.FieldDescriptorProto.Type
+  defp type_from_field_props(%Protobuf.FieldProps{type: type}) do
+    case type do
+      :double ->
+        {1, nil}
+
+      :float ->
+        {2, nil}
+
+      :int64 ->
+        {3, nil}
+
+      :uint64 ->
+        {4, nil}
+
+      :int32 ->
+        {5, nil}
+
+      :fixed64 ->
+        {6, nil}
+
+      :fixed32 ->
+        {7, nil}
+
+      :bool ->
+        {8, nil}
+
+      :string ->
+        {9, nil}
+
+      :group ->
+        {10, nil}
+
+      :message ->
+        {11, "." <> to_string(type)}
+
+      :bytes ->
+        {12, nil}
+
+      :uint32 ->
+        {13, nil}
+
+      :enum ->
+        {14, nil}
+
+      :sfixed32 ->
+        {15, nil}
+
+      :sfixed64 ->
+        {16, nil}
+
+      :sint32 ->
+        {17, nil}
+
+      :sint64 ->
+        {18, nil}
+
+      _ ->
+        if is_atom(type) and Code.ensure_loaded?(type) and
+             function_exported?(type, :descriptor, 0) do
+          {11, get_pb_type_name(type)}
+        else
+          raise("Unsupported type")
+        end
```

Comment:
There appears to be no test covering this case.  How is this triggered?  Can we have test coverage for this?

If this cannot be covered and is "just in case" code I'd prefer it be removed -- we wont be able to maintain it in a meaningful way

Reply:
Added test cases for this.